Menus/InventoryMenuGUI.cpp

```diff
@@ -0,0 +1,1237 @@
+				}
+			}
+			ClearSelectedItem();
+		}
+		pressedButton->OnLoseFocus();
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void InventoryMenuGUI::SwapEquippedItemAndInventoryItem(MovableObject *equippedItemToSwapOut, int inventoryItemIndexToSwapIn) {
+		MovableObject *offhandEquippedItemToAddToInventory = nullptr;
+		if (m_InventoryActorEquippedItems.size() > 1 && inventoryItemIndexToSwapIn >= 0 && inventoryItemIndexToSwapIn < m_InventoryActor->GetInventorySize()) {
+			const HeldDevice *inventoryItemToSwapIn = dynamic_cast<const HeldDevice *>(m_InventoryActor->GetInventory()->at(inventoryItemIndexToSwapIn));
+			if (m_InventoryActorEquippedItems.at(0) && !inventoryItemToSwapIn->IsOneHanded() && !inventoryItemToSwapIn->HasObjectInGroup("Shields")) {
+				equippedItemToSwapOut = m_InventoryActorEquippedItems.at(0);
+				offhandEquippedItemToAddToInventory = m_InventoryActorEquippedItems.at(1);
+			}
+		}
+
+		Arm *equippedItemArm = dynamic_cast<Arm *>(equippedItemToSwapOut->GetParent());
+		equippedItemArm->SetHeldMO(m_InventoryActor->SetInventoryItemAtIndex(equippedItemArm->ReleaseHeldMO(), inventoryItemIndexToSwapIn));
+		if (offhandEquippedItemToAddToInventory) { m_InventoryActor->AddInventoryItem(dynamic_cast<Arm *>(offhandEquippedItemToAddToInventory->GetParent())->ReleaseHeldMO()); }
+		m_InventoryActor->GetDeviceSwitchSound()->Play(m_MenuController->GetPlayer());
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void InventoryMenuGUI::ReloadSelectedItem() {
+		if (!m_InventoryActorIsHuman) {
+			return;
+		}
+		const AHuman *inventoryActorAsAHuman = dynamic_cast<AHuman *>(m_InventoryActor);
+		if (m_GUISelectedItem == nullptr) {
+			inventoryActorAsAHuman->ReloadFirearms();
+		} else if (HDFirearm *selectedItemObjectAsFirearm = dynamic_cast<HDFirearm *>(m_GUISelectedItem->Object)) {
+			if (m_GUISelectedItem->InventoryIndex > -1) {
+				if (!m_InventoryActorEquippedItems.empty()) {
+					SwapEquippedItemAndInventoryItem(m_InventoryActorEquippedItems.at(0), m_GUISelectedItem->InventoryIndex);
+				} else if (inventoryActorAsAHuman->GetFGArm() || inventoryActorAsAHuman->GetBGArm()) {
+					Arm *armToUse = inventoryActorAsAHuman->GetFGArm() ? inventoryActorAsAHuman->GetFGArm() : inventoryActorAsAHuman->GetBGArm();
+					armToUse->SetHeldMO(m_InventoryActor->RemoveInventoryItemAtIndex(m_GUISelectedItem->InventoryIndex));
+				}
+			}
+			selectedItemObjectAsFirearm->Reload();
+		}
+		ClearSelectedItem();
+		m_GUIReloadButton->OnLoseFocus();
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void InventoryMenuGUI::DropSelectedItem(const Vector *dropDirection) {
+		auto LaunchInventoryItem = [this, &dropDirection](MovableObject *itemToLaunch) {
```

Comment:
I wanna stop doing comments on lambdas tbh, unless they're unclear. I think the name of the lambda explains this as much as any comment would, though I can change it to be more informative - something like `LaunchInventoryItemBasedOnDropDirection`.

Reply:
That's essentially not commenting on class methods because their names are clear.
This (and many others) doesn't even have to be a lambda, but you chose to lambda it anyway, so add comment. ¯\\\_( ͡° ͜ʖ ͡°)_/¯

Reply:
Main difference between lambdas and methods is that all these lambdas are only ever used in one very particular scope, basically as a way to avoid duplicate code. Point of method comments is to explain to external users what a method does, right? This has no external users, it's only ever supposed to be used inside this method. If someone wanted to use it somewhere else, I'd say it should be turned into a private method or whatever, with a comment.
Ultimately it's not a huge deal, if you super care about it then I'll do it, but to me it just seems kinda silly/pointless to do method comments on lambdas.